yarn-project/circuit-types/src/tx/processed_tx.ts

```diff
@@ -36,37 +25,44 @@ export enum PublicKernelPhase {
  * Represents a tx that has been processed by the sequencer public processor,
  * so its kernel circuit public inputs are filled in.
  */
-export type ProcessedTx = Pick<Tx, 'clientIvcProof' | 'noteEncryptedLogs' | 'encryptedLogs' | 'unencryptedLogs'> & {
-  /**
-   * Output of the private tail or public tail kernel circuit for this tx.
-   */
-  data: KernelCircuitPublicInputs;
+export type ProcessedTx = {
   /**
    * Hash of the transaction.
    */
   hash: TxHash;
   /**
-   * Flag indicating the tx is 'empty' meaning it's a padding tx to take us to a power of 2.
+   * Tx.data. The output of the private kernel tail or tail_to_public.
    */
-  isEmpty: boolean;
+  data: PrivateKernelTailCircuitPublicInputs;
   /**
-   * Reason the tx was reverted.
+   * Proof from the private kernel circuit.
    */
-  revertReason: SimulationError | undefined;
+  clientIvcProof: ClientIvcProof;
   /**
    * The request for AVM proving.
    */
   avmProvingRequest: AvmProvingRequest | undefined;
+  /**
+   * TxConstantData specified by the user combined with global variables.
+   */
+  constants: CombinedConstantData;
+  /**
+   * Output data of the tx.
+   */
+  txEffect: TxEffect;
```

Comment:
`data: KernelCircuitPublicInputs` is removed from `ProcessedTx` because it is no longer the type exported by all txs. Private-only txs still output `KernelCircuitPublicInputs`. But txs containing public calls are now associated with `AvmCircuitPublicIpnuts`.

We replace it with `TxEffect`, which achieves what `KernelCircuitPublicInputs` can do and more:
- The `finalPublicDataUpdateRequests` field can be removed because `txEffect.publicDataWrites` contains all public data writes, including the `feePaymentPublicDataWrite` injected by the base rollup.
- All 3 log preimages arrays can be removed because they are already in `TxEffect`.